src/Command/CodeStudio/CodeStudioSetCICDVarCommand.php

```diff
@@ -0,0 +1,213 @@
+        $project_choice_labels,
+        $project_choice_labels[0],
+      );
+      $choice_id = $this->io->askQuestion($question);
+      $project_id = array_search($choice_id, $list, TRUE);
+    }
+
+    // List of CI/CD variables.
+    $variables_list = $this->getSupportedVariables();
+    $variable_names = array_keys($variables_list);
+
+    $variable = NULL;
+    if ($input->getOption('gitlab-cicd-var')) {
+      $variable = $this->input->getOption('gitlab-cicd-var');
+      // If job provided by user is not valid.
+      if (!in_array($variable, $variable_names)) {
+        $this->io->error('Variable provided is not valid.');
+        return 1;
+      }
+    }
+
+    if (!$variable) {
+      $question = new ChoiceQuestion('Please select the Code Studio CI/CD variable you want to update.',
+        $variable_names,
+        $variable_names[0],
+      );
+      $variable = $this->io->askQuestion($question);
+    }
+
+    // Check if the variable already exists or not.
+    $all_project_variables = $this->gitLabClient->projects()->variables($project_id);
+    $is_variable_exist = FALSE;
+    foreach (array_values($all_project_variables) as $project_variable) {
+      if ($project_variable['key'] === $variable) {
+        $is_variable_exist = TRUE;
+        break;
+      }
+    }
+
+    $variable_value = '';
+    if ($this->input->getOption('gitlab-cicd-var-value')) {
+      // If user has provided the variable value as an option.
+      $variable_value = $this->input->getOption('gitlab-cicd-var-value');
+    }
+
+    // If variable value is not provided, ask user to enter it.
+    if (!$variable_value) {
+      $variable_val_question = new Question('Please enter the CI/CD variable value');
+      $variable_value = $this->io->askQuestion($variable_val_question);
+    }
+
+    try {
+      // If  variable not exists in project, create it.
+      if (!$is_variable_exist) {
+        $this->gitLabClient->projects()->addVariable($project_id, $variable, $variable_value);
+      } else {
+        // If variable already exists, updating the variable.
+        $this->gitLabClient->projects()->updateVariable($project_id, $variable, $variable_value);
+      }
+    }
+    catch (RuntimeException $exception) {
+      $this->io->error([
+        "Unable to update the job",
+      ]);
+      // Log the error for debugging purpose.
+      $this->logger->debug('Error @error while updating/creating the job @variable for the project @project project id @project_id', [
+        '@error' => $exception->getMessage(),
+        '@project' => $list[$project_id],
+        '@project_id' => $project_id,
+        '@variable' => $variable,
+      ]);
+      return 1;
+    }
+
+    $this->io->info("Job `{$variable}` is updated for the project `{$list[$project_id]}` with value '{$variable_value}'");
+    return 0;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  protected function getGitLabHost(): string {
```

Comment:
Duplicate

Reply:
Here we are overriding the method of parent class to support/add the `gitlab-host-name` param which is not there in parent class (to make it generic)